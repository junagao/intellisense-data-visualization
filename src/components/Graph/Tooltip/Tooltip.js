import React from 'react';
import PropTypes from 'prop-types';

import { TooltipDiv, TooltipH1, TooltipP, TooltipSpan } from './Tooltip.styles';

const Tooltip = ({ x, y, color, metric, time, value }) => (
  <TooltipDiv x={x} y={y}>
    <TooltipH1 color={color}>{metric}</TooltipH1>
    {time && value ? (
      <>
        <TooltipP>
          <TooltipSpan>time:</TooltipSpan> {time}
        </TooltipP>
        <TooltipP>
          <TooltipSpan>value:</TooltipSpan> {value}
        </TooltipP>
      </>
    ) : null}
  </TooltipDiv>
);

Tooltip.propTypes = {
  color: PropTypes.string.isRequired,
  metric: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
};

export default Tooltip;
