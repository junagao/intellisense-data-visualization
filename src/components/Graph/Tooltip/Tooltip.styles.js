import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`;

export const TooltipDiv = styled.div.attrs(({ x, y }) => ({
  style: {
    marginLeft: `${x - 20}px`,
    marginTop: `${y + 25}px`,
  },
}))`
  position: absolute;
  animation: 0.2s ${fadeIn} ease-in;
  background-color: #fff;
  border-radius: 0.25rem;
  box-shadow: 0 1px 4px 0 #eaeaea, 0 0 0 1px #eaeaea;
  font-size: 0.65rem;
  padding: 0.5rem;
`;

export const TooltipH1 = styled.h1.attrs(({ color }) => ({ style: { color } }))`
  font-size: 0.7rem;
  font-weight: 500;
`;

export const TooltipP = styled.p`
  color: #333;
  opacity: 0.85;
  :first-of-type {
    margin-top: 0.25rem;
  }
`;

export const TooltipSpan = styled.span`
  opacity: 0.75;
`;
