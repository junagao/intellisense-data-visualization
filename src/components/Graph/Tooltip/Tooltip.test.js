import React from 'react';
import { shallow } from 'enzyme';

import Tooltip from './Tooltip';

describe('Tooltip', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Tooltip
        x={12}
        y={23}
        color="red"
        metric="metric A"
        time="20"
        value="2342"
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one TooltipDiv element', () => {
    expect(wrapper.find('TooltipDiv').length).toEqual(1);
  });

  it('should render one TooltipH1 element with the metric name and pass the `color` prop', () => {
    expect(wrapper.find('TooltipH1').length).toEqual(1);
    expect(wrapper.find('TooltipH1').prop('color')).toEqual('red');
    expect(wrapper.find('TooltipH1').at(0).text()).toEqual('metric A');
  });

  it('should render one TooltipP element with one TooltipSpan element with the time and another TooltipP element with a TooltipSpan with the value  ', () => {
    expect(wrapper.find('TooltipP').length).toEqual(2);
    expect(wrapper.find('TooltipSpan').length).toEqual(2);
    expect(wrapper.find('TooltipSpan').at(0).text()).toEqual('time:');
    expect(wrapper.find('TooltipP').at(0).childAt(2).text()).toEqual('20');
    expect(wrapper.find('TooltipSpan').at(1).text()).toEqual('value:');
    expect(wrapper.find('TooltipP').at(1).childAt(2).text()).toEqual('2342');
  });

  it('should render only the TooltipH1 element if time and value is an empty string - showing the metric and TooltipDiv having only the `x, y` props', () => {
    wrapper.setProps({ time: '', value: '' });
    expect(wrapper.find('TooltipDiv').prop('x')).toEqual(12);
    expect(wrapper.find('TooltipDiv').prop('y')).toEqual(23);
    expect(wrapper.find('TooltipH1').length).toEqual(1);
    expect(wrapper.find('TooltipH1').prop('color')).toEqual('red');
    expect(wrapper.find('TooltipH1').at(0).text()).toEqual('metric A');
    expect(wrapper.find('TooltipP').length).toEqual(0);
  });
});
