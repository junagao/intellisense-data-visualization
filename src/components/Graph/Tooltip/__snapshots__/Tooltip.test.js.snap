// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render correctly given the required props 1`] = `
<TooltipDiv
  x={12}
  y={23}
>
  <TooltipH1
    color="red"
  >
    metric A
  </TooltipH1>
  <TooltipP>
    <TooltipSpan>
      time:
    </TooltipSpan>
     
    20
  </TooltipP>
  <TooltipP>
    <TooltipSpan>
      value:
    </TooltipSpan>
     
    2342
  </TooltipP>
</TooltipDiv>
`;
